File: src/ServerPulse.Backend/AnalyzerApi.Infrastructure/Dtos/Responses/Statistics/SlotStatisticsResponse.cs
using AnalyzerApi.Infrastructure.Dtos.Responses.Events;

namespace AnalyzerApi.Infrastructure.Dtos.Responses.Statistics
{
    public class SlotStatisticsResponse
    {
        public DateTime CollectedDateUTC { get; set; }
        public ServerLifecycleStatisticsResponse? GeneralStatistics { get; set; }
        public ServerLoadStatisticsResponse? LoadStatistics { get; set; }
        public ServerCustomStatisticsResponse? CustomEventStatistics { get; set; }
        public IEnumerable<LoadEventResponse> LastLoadEvents { get; set; } = [];
        public IEnumerable<CustomEventResponse> LastCustomEvents { get; set; } = [];
    }
}

File: src/ServerPulse.Frontend/src/app/modules/analyzer/domain/dtos/responses/statistics/slotStatisticsResponse.ts
import { CustomEventResponse, LoadEventResponse, mapCustomEventResponseToCustomEvent, mapLoadEventResponseToLoadEvent, mapServerCustomStatisticsResponseToServerCustomStatistics, mapServerLifecycleStatisticsResponseToServerLifecycleStatistics, mapServerLoadStatisticsResponseToServerLoadStatistics, ServerCustomStatisticsResponse, ServerLifecycleStatisticsResponse, ServerLoadStatisticsResponse, SlotStatistics } from "../../../..";

export interface SlotStatisticsResponse {
    collectedDateUTC: Date,
    generalStatistics: ServerLifecycleStatisticsResponse | null,
    loadStatistics: ServerLoadStatisticsResponse | null,
    customEventStatistics: ServerCustomStatisticsResponse | null,
    lastLoadEvents: LoadEventResponse[],
    lastCustomEvents: CustomEventResponse[],
}

export function mapSlotStatisticsResponseToSlotStatistics(response: SlotStatisticsResponse): SlotStatistics {
    return {
        collectedDateUTC: response?.collectedDateUTC,
        generalStatistics: response.generalStatistics === null ? null : mapServerLifecycleStatisticsResponseToServerLifecycleStatistics(response.generalStatistics),
        loadStatistics: response.loadStatistics === null ? null : mapServerLoadStatisticsResponseToServerLoadStatistics(response.loadStatistics),
        customEventStatistics: response.customEventStatistics === null ? null : mapServerCustomStatisticsResponseToServerCustomStatistics(response.customEventStatistics),
        lastLoadEvents: response.lastLoadEvents.map(mapLoadEventResponseToLoadEvent),
        lastCustomEvents: response.lastCustomEvents.map(mapCustomEventResponseToCustomEvent),
    };
}